src/octopeer-github/test/ElementSelectionBehaviour/ElementSelectionBehaviourTest.ts

```diff
@@ -0,0 +1,182 @@
+];
+
+let filesChangedSelectors: CreatableNumberPair[] = [
+    [AddEmoticonButtonElementSelectionBehaviour, 1],
+    [CancelEditPRNameButtonElementSelectionBehaviour, 0],
+    // [CancelInlineCommentButtonElementSelectionBehaviour, 0],
+    // [CommentInlineCommentButtonElementSelectionBehaviour, 0],
+    [EditCommentButtonElementSelectionBehaviour, 1],
+    [EditPRNameButtonElementSelectionBehaviour, 1],
+    // [InlineCommentButtonElementSelectionBehaviour, 0],
+    [SaveEditPRNameButtonElementSelectionBehaviour, 0],
+    [DateMiscellaneousElementSelectionBehaviour, 5],
+    // [OtherContributerNameElementSelectionBehaviour, 0],
+    // [PRCreatorNameElementSelectionBehaviour, 0],
+    // [PRParticipantNameElementSelectionBehaviour, 1],
+    [CommitsTabHeaderElementSelectionBehaviour, 1],
+    [ConversationTabHeaderElementSelectionBehaviour, 1],
+    [FilesChangedTabHeaderElementSelectionBehaviour, 1],
+];
+
+/**
+ * This is a list of the different html files that are used to test the classes.
+ * htmlList[x] corresponds with selectorListsList[x].
+ * @type {string[]}
+ */
+const htmlsList = [
+    "conversation-tab.html",
+    "commit-tab.html",
+    "files-changed-tab.html",
+];
+
+/**
+ * This is a list of the different CreatableNumberPairLists that are used to test the classes.
+ * selectorListsList[x] corresponds with htmlList[x].
+ * @type {CreatableNumberPair[][]}
+ */
+const selectorListsList = [
+    conversationSelectors,
+    commitSelectors,
+    filesChangedSelectors,
+];
+
+/**
+ * Strips the Creatables, so only the name remains. Used for logging the test name.
+ * @param fun
+ * @returns {string}
+ */
+function functionName(fun: ElementSelectionBehaviourCreatable) {
+    let ret = fun.toString();
+    ret = ret.substr("function ".length);
+    ret = ret.substr(0, ret.indexOf("("));
+    return ret;
+}
+
+let databaseSpy: jasmine.Spy;
+jasmine.getFixtures().fixturesPath = "base/src/octopeer-github/test/resources";
+
+/**
+ * Every htmlFile is paired with the corresponding CreatableNumberPairList. The Creatables in the list
+ * have a determined amount of times they should appear in the html file. Besides these tests, it is also
+ * tested whether the database was called to log the data.
+ */
+for (let i = 0; i < htmlsList.length; i++) {
+    const htmlFile = htmlsList[i];
+    const selectors = selectorListsList[i];
+
+    for (let j = 0; j < selectors.length; j++) {
+        const selectorCreatable = selectors[j][0];
+        describe("A " + functionName(selectorCreatable) + " that selects Elements at " + htmlFile, function () {
+            const database = new ConsoleLogDatabaseAdapter();
+            const selector = new selectorCreatable(database);
+
+            beforeEach(function () {
+                databaseSpy = spyOn(database, "post");
+                loadFixtures(htmlFile);
+            });
+
+            it("should select exactly the specified amount of Elements", function () {
```

Comment:
Please put this "specific amount of elements" in the description of the test, like with the class names and html files.